test/correctness/predicated_store_load.cpp

```diff
@@ -464,6 +464,22 @@ int vectorized_predicated_load_lut_test(const Target &t) {
     return 0;
 }
 
+int predicated_atomic_store_test(const Target &t) {
+    // We don't support atomic predicated stores, so ensure that we don't
+    // generate them. See https://github.com/halide/Halide/issues/8280
+    ImageParam in(Float(32), 1);
+    Func f;
+    Var x;
+    RDom r(0, 20);
+
+    f(x) = 0.f;
+    f(x) += in(r) + x;
+    f.update().vectorize(x, 8, TailStrategy::GuardWithIf).atomic().parallel(r);
+
+    f.compile_jit(t);
+    return 0;
```

Comment:
...this can never return anything but 0, so the test can never fail. Are we missing something here?

Reply:
The test can throw an internal error in compile_jit, which is what it used to do.

Reply:
(I'll add a comment)